### PR TITLE
Disable analytics feature by default

### LINKED ISSUE #857: Verify health-check after disabling redis-store
_From @channing-jackson on August 26, 2016 17:57_

Deployed 1.0.0.RC1 successfully, and checked /management/health endpoint and it shows down. 

Following are the entries in the manifest: 

```
applications: 
- name: mydataflow
  memory: 1024M
  disk_quota: 1024M
  no-route: false
  no-hostname: false
  health-check-type: 'port' 
  buildpack: custom_java_buildpack_offline
  env: 
    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_URL: https://api.dev.xyz.com
    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_ORG: myorg
    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_SPACE: myspace
    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_DOMAIN: dev.xyz.com
    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_STREAM_SERVICES: dev-bde-rabbit,dev-bde-sccs
    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_USERNAME: <user>
    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_PASSWORD: <password>
    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_SKIP_SSL_VALIDATION: true
    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_STREAM_BUILDPACK: custom_java_buildpack_offline
    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_STREAM_ENABLE_RANDOM_APP_NAME_PREFIX: false
    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_STREAM_APP_NAME_PREFIX: abc
    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_STREAM_MEMORY: 768
    JAVA_OPTS: -Dhttp.keepAlive=false
    MAVEN_REMOTE_REPOSITORIES_SNAPSHOTS_URL: https://mavenrepo.xyz.com/nexus/content/repositories/snapshots/
    MAVEN_REMOTE_REPOSITORIES_RELEASES_URL: https://mavenrepo.xyz.com/nexus/content/repositories/releases/
    MAVEN_REMOTE_REPOSITORIES_SPRING_MILESTONES_URL: https://mavenrepo.xyz.com/nexus/content/repositories/spring-milestones/
    MAVEN_REMOTE_REPOSITORIES_SPRING_SNAPSHOTS_URL: https://mavenrepo.xyz.com/nexus/content/repositories/spring-snapshots/
    MAVEN_REMOTE_REPOSITORIES_SPRING_RELEASE_URL: https://mavenrepo.xyz.com/nexus/content/repositories/spring-release/
  services: 
    - dev-bde-rabbit
    - dev-bde-mysql
    - dev-bde-sccs
```
Application is not actually down.  It is running fine.

_Copied from original issue: spring-cloud/spring-cloud-dataflow-server-cloudfoundry#169_

### PR DESCRIPTION
**Pull request by ilayaperumalg**
- Set default analytics feature to be 'false'
 - Override redis health indicator to only check health if analytics configuration is enabled otherwise it will return 'UNKNOWN'
   - this will avoid false info on overall app status to be 'DOWN'

Fixes #857

### CODE REVIEW

File: spring-cloud-dataflow-docs/src/main/asciidoc/getting-started.adoc [hunk 1]
```diff
@@ -36,7 +36,7 @@ One can enable, disable these features by setting the following boolean properti
 * `spring.cloud.dataflow.features.tasks-enabled`
 * `spring.cloud.dataflow.features.analytics-enabled`
 
-By default, all these features are enabled for `local` dataflow server.
+By default, all these features except 'analytics' features are enabled.
```
Comment (markfisher) [inline spring-cloud-dataflow-docs/src/main/asciidoc/getting-started.adoc]: don't need "features" 2x in this sentence (probably best to remove the 2nd occurrence)

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/features/FeaturesConfiguration.java [hunk 2]
```diff
@@ -34,15 +37,43 @@
  * @author Ilayaperumal Gopinathan
  */
 @Configuration
-@Import({ AnalyticsConfiguration.class, StreamConfiguration.class, TaskConfiguration.class })
+@Import({ AnalyticsConfiguration.class, StreamConfiguration.class,
+		TaskConfiguration.class })
 public class FeaturesConfiguration {
 
+	@Autowired()
```
Comment (markfisher) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/features/FeaturesConfiguration.java]: don't need the parens here

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/features/FeaturesProperties.java [hunk 3]
```diff
@@ -33,7 +33,7 @@
 
     public static final String ANALYTICS_ENABLED = "analytics-enabled";
 
-    private boolean analyticsEnabled = true;
+    private boolean analyticsEnabled;
```
Comment (markfisher) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/features/FeaturesProperties.java]: maybe explicitly setting to `false` would be more clear (from a self-documenting perspective)

File: spring-cloud-dataflow-server-local/src/test/java/org/springframework/cloud/dataflow/server/local/LocalConfigurationTests.java [hunk 4]
```diff
@@ -18,9 +18,7 @@
 
 import static org.hamcrest.Matchers.instanceOf;
 import static org.hamcrest.Matchers.is;
-import static org.junit.Assert.assertNotNull;
-import static org.junit.Assert.assertThat;
-import static org.junit.Assert.fail;
+import static org.junit.Assert.*;
```
Comment (markfisher) [inline spring-cloud-dataflow-server-local/src/test/java/org/springframework/cloud/dataflow/server/local/LocalConfigurationTests.java]: avoid wildcard imports

**Comment by ilayaperumalg** [comment]
Pushed the changes to address review comments and rebased the PR branch as well.

**Comment by markfisher** [comment]
LGTM; rebased, squashed, and merged as https://github.com/spring-cloud/spring-cloud-dataflow/commit/4678461ef15a02948516a1510a16a9fa64294a39

**Pull request closed without merging**